src/main/java/org/mule/extension/email/internal/StoredEmailContentFactory.java

```diff
@@ -134,4 +162,27 @@ private static MediaType getMediaType(Message message) {
       return MediaType.TEXT;
     }
   }
+
+  private String getUniqueFileName(Set<String> keys, String fileName) {
+    if (keys.contains(fileName)) {
+      return resolveUniqueFileName(keys, fileName, 0);
+    }
+    return fileName;
+  }
+
+  private String resolveUniqueFileName(Set<String> keys, String fileName, Integer lastSuffixTried) {
+    String extension = "";
+    String baseName = fileName;
+    int lastIndexOf = fileName.lastIndexOf('.');
+    if (lastIndexOf != -1) {
+      extension = fileName.substring(fileName.lastIndexOf('.'));
```

Comment:
use the previously defined variable instead of calling the lastIndexOf method again